### ISSUE TITLE
exp/orderbook: Invalid last price level amount in /order_book

### ISSUE THREAD

**Issue by bartekn** [stellar/go#2232]
### What version are you using?

Horizon v1.0.0-beta.

### What did you do?

`horizon-cmp` found the following diff at ledger 28100122 in:
```
https://horizon-stg-pubnet.stellar.org/order_book?selling_asset_type=credit_alphanum4&selling_asset_code=LFEC&selling_asset_issuer=GAG6FS3CR64QJHLHJU7HNXUB4KBLXVDFQBDXM5LG22WOM7CA2ITJAVD2&buying_asset_type=native&limit=50&c=0.6118116069996031
```
```diff
> {"price_r":{"n":1,"d":100000},"price":"0.0000100","amount":"10.0000000"}],
< {"price_r":{"n":1,"d":100000},"price":"0.0000100","amount":"1.0000000"}],
```

I think it can connected to the issue we are already aware of https://github.com/stellar/go/issues/2010 and the code responsible is here:

https://github.com/stellar/go/blob/db27f8d0a7dc2545ad174cc7890b323be648fe8e/exp/orderbook/graph.go#L178-L183

It's possible that `results[len(results)-1].Price` and `offer.Price` is actually the same price but expressed using different N and D (ex. `1/100` = `10/1000`). This makes some offers not being included in the price level.

**Comment by bartekn** [comment]
Not critical so moving to next milestone. @ire-and-curses

**Comment by bartekn** [comment]
Found another instance of this issue:

```diff
1c1
< https://horizon.stellar.org /order_book?selling_asset_type=credit_alphanum4&selling_asset_code=LFEC&selling_asset_issuer=GAG6FS3CR64QJHLHJU7HNXUB4KBLXVDFQBDXM5LG22WOM7CA2ITJAVD2&buying_asset_type=native&limit=50&c=0.6118116069996031
---
> https://horizon-stg-pubnet.stellar.org /order_book?selling_asset_type=credit_alphanum4&selling_asset_code=LFEC&selling_asset_issuer=GAG6FS3CR64QJHLHJU7HNXUB4KBLXVDFQBDXM5LG22WOM7CA2ITJAVD2&buying_asset_type=native&limit=50&c=0.6118116069996031
328,329c328,329
<         "n": 1,
<         "d": 500
---
>         "n": 20000,
>         "d": 10000000
332c332
<       "amount": "1.8999500"
---
>       "amount": "31.4999950"
336,337c336,337
<         "n": 20000,
<         "d": 10000000
---
>         "n": 1,
>         "d": 500
340c340
<       "amount": "31.4999950"
---
>       "amount": "1.8999500"
381,396d380
<     },
<     {
<       "price_r": {
<         "n": 1000,
<         "d": 10000000
<       },
<       "price": "0.0001000",
<       "amount": "500.0000000"
<     },
<     {
<       "price_r": {
<         "n": 101,
<         "d": 10000000
<       },
<       "price": "0.0000101",
<       "amount": "20.0000000"
```

https://gist.github.com/bartekn/cfef9ec4d707a5d61f244fe45de3e774

**Comment by 2opremio** [comment]
I am pretty sure that the last amount is due to wrongly-computed price-level buckets (fixed by #2400 ). The problem goes away if you increase the `limit` parameter to a large-enough value, which indicates that the problem comes from counting price levels incorrectly.
